src/gc/gc.d

```diff
@@ -1543,6 +1519,13 @@ struct Gcx
      */
     void runFinalizers(in void[] segment) nothrow
     {
+        GC.inFinalizer = true;
+        scope (failure) // unlock on finalize errors
+        {
+            GC.inFinalizer = false;
+            GC.gcLock.unlock();
```

Comment:
This is confusing - it doesn't pair with a visible lock call in the same scope. Couldn't the same entity that locked also take care of the unlock?